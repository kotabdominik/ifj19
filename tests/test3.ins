číselný výraz (48-16)*4+4-(78//5*5) se rovná
