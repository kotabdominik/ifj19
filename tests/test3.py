a = (48-16)*4+4-(78//5*5)
print(a)