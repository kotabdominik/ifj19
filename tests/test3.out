.IFJcode19
JUMP $$MAIN
LABEL $ORD
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$POSIT
POPS LF@$POSIT
DEFVAR LF@$STRING
POPS LF@$STRING
DEFVAR LF@$CHR2INT
DEFVAR LF@$TMPLEN
STRLEN LF@$TMPLEN LF@$STRING
ADD LF@$TMPLEN LF@$TMPLEN int@-1
DEFVAR LF@$BOOLCHECK
GT LF@$BOOLCHECK LF@$POSIT LF@$TMPLEN
JUMPIFEQ $ERRORORD LF@$BOOLCHECK bool@true
LT LF@$BOOLCHECK LF@$POSIT int@0
JUMPIFEQ $ERRORORD LF@$BOOLCHECK bool@true
ADD LF@$TMPLEN LF@$TMPLEN int@1
STRI2INT LF@$RETVAL LF@$STRING LF@$POSIT
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ERRORORD
MOVE LF@$RETVAL string@None
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $BETA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL int
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $SUBSTR
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$DELKA
POPS LF@$DELKA
DEFVAR LF@$POSIT
POPS LF@$POSIT
DEFVAR LF@$STRING
POPS LF@$STRING
DEFVAR LF@$TMPLEN
STRLEN LF@$TMPLEN LF@$STRING
DEFVAR LF@$BOOLCHECK
ADD LF@$TMPLEN LF@$TMPLEN int@-1
GT LF@$BOOLCHECK LF@$POSIT LF@$TMPLEN
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
LT LF@$BOOLCHECK LF@$POSIT int@0
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
GT LF@$BOOLCHECK int@0 LF@$DELKA
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
EQ LF@$BOOLCHECK int@0 LF@$DELKA
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
ADD LF@$TMPLEN LF@$TMPLEN int@1
GETCHAR LF@$RETVAL LF@$STRING LF@$POSIT
ADD LF@$POSIT LF@$POSIT int@1
DEFVAR LF@$TMPSTR
ADD LF@$DELKA LF@$DELKA LF@$POSIT
SUB LF@$DELKA LF@$DELKA int@1
LABEL $WHILE_LOOP_B
JUMPIFEQ $WHILE_LOOP_E LF@$POSIT LF@$DELKA
JUMPIFEQ $WHILE_LOOP_E LF@$POSIT LF@$TMPLEN
GETCHAR LF@$TMPSTR LF@$STRING LF@$POSIT
CONCAT LF@$RETVAL LF@$RETVAL LF@$TMPSTR
ADD LF@$POSIT LF@$POSIT int@1
JUMP $WHILE_LOOP_B
LABEL $WHILE_LOOP_E
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ERRORSUB
MOVE LF@$RETVAL string@None
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ALFA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL string
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $GAMA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL float
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $LENGTH
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$1
POPS LF@$1
STRLEN LF@$RETVAL LF@$1
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $CHAR
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$ASC
POPS LF@$ASC
INT2CHAR LF@$RETVAL LF@$ASC
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ZEROERROR
EXIT int@9
LABEL $$MAIN
CREATEFRAME
PUSHFRAME
DEFVAR GF@$VARa
PUSHS int@48
PUSHS int@16
SUBS
PUSHS int@4
MULS
PUSHS int@4
ADDS
PUSHS int@78
PUSHS int@5
CREATEFRAME
PUSHFRAME
DEFVAR LF@$1
DEFVAR LF@$2
DEFVAR LF@$VAL1
DEFVAR LF@$VAL2
DEFVAR LF@$REESULT
POPS LF@$VAL2
POPS LF@$VAL1
MOVE LF@$1 LF@$VAL1
MOVE LF@$2 LF@$VAL2
JUMPIFEQ $ZEROERROR int@0 LF@$2
IDIV LF@$REESULT LF@$VAL1 LF@$VAL2
PUSHS LF@$REESULT
POPFRAME
PUSHS int@5
MULS
SUBS
POPS GF@$VARa
PUSHS GF@$VARa
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL0x7ffff2fb67d4
DEFVAR LF@$PEPEGA0
POPS LF@$PEPEGA0
WRITE LF@$PEPEGA0
WRITE string@\010
MOVE LF@$RETVAL0x7ffff2fb67d4  string@None
POPFRAME
